framework/src/controller/bus.ts

```diff
@@ -225,18 +240,33 @@ export class Bus {
 		});
 	}
 
-	public publish(eventName: string, eventValue?: object): void {
+	public publish(eventData: string | JSONRPC.NotificationRequest): void {
+		const parsedEvent = Event.fromJSONRPCNotification(eventData);
+
+		try {
+			JSONRPC.validateJSONRPCRequest(parsedEvent.toJSONRPCNotification() as never);
+		} catch (error) {
+			this.logger.error(
+				JSONRPC.errorResponse(null, JSONRPC.invalidRequest()),
+				'Invalid publish request',
```

Comment:
same here

Reply:
The error handling is improved in upcoming PR.